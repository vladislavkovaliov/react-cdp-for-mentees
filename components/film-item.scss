@import '../constants/_variables.scss';

.film-item {
  max-width: 192px;
  width: 100%;
  position: relative;

  margin: {
    right: 14px;
    bottom: 8px;
  };

  .rating {
    position: absolute;
    right: 8px;
    top: 8px;
    color: white;
    background-color: rgba(53, 191, 0, 0.7);
    border-radius: 50%;
    height: 32px;
    width: 32px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &__img {
    background: {
      position: 50%;
      size: cover;
      repeat: no-repeat;
    };
    height: 280px;
    width: 100%;
  }

  &__footer {
    &-desc {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      margin-top: 4px;
      text-transform: uppercase;
      align-items: center;

      span {
        border: 1px solid $grey;
        border-radius: 4px;
        padding: 2px;
      }
    }

    &-desc,
    &-gender {
      font: {
        size: 12px;
      }
    }

    &-gender {
      font: {
        weight: 300;
      }
    }
  }
}