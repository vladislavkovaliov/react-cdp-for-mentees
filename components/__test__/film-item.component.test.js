import React from 'react';
import { FilmItem } from '../film-item.component';

import shallow from 'enzyme/shallow';

describe('<FilmItem />', () => {
  test('Shapshot <FilmItem />', () => {
    const wrapper = shallow(
      <FilmItem
        releaseDate="2007-12-12"
        genres={[ 'drama', 'western' ]}
        vote_average="4.6"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
