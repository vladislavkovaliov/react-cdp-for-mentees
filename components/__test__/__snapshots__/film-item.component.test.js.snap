// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmItem /> Shapshot <FilmItem /> 1`] = `
<React.Fragment>
  <div
    className="film-item"
  >
    <span
      className="rating"
    />
    <div
      className="film-item__img"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    />
    <div
      className="film-item__footer"
    >
      <div
        className="film-item__footer-desc"
      >
        <Link
          href={
            Object {
              "pathname": "/film-info",
              "query": Object {
                "id": undefined,
              },
            }
          }
        >
          <p />
        </Link>
        <span>
          2007
        </span>
      </div>
      <div
        className="film-item__footer-gender"
      >
        drama & western
      </div>
    </div>
  </div>
</React.Fragment>
`;
